### DISCUSSION TITLE
Allure generate empty report

### DISCUSSION THREAD

**Discussion by huynguyen-anduin** [allure-framework/allure2#2585] [Questions & Support] [answered]
I have a folder of raw allure-results, contains reports of 2 days. When I use `allure generate --single-file`, but when open the html file on browser:

1. With reports of 2 days:

![image](https://github.com/allure-framework/allure2/assets/141135676/f5813dc9-d6bd-49fa-9506-30d332847276)

2. With reports of day 1st: Generate report correctly

3. With reports of day 2nd: Generate report correctly

What I tried:

1. Checked all json files, they are in correct format.
2. `allure serve` -> the report looks correct.
3. `allure generate allure-results -o  report` (without --single-file) then `allure open` ->  the report looks correct.

The raw reports are generated by allure-testng version 2.24.0. I tried allure 2.24.0, 2.24.1 and 2.29.0 to generate single html.

What maybe the reason here?

**Comment by baev** [comment]
Could you please check the browser console logs for errors?

**Comment by huynguyen-anduin** [comment]
@baev Thanks for your response.

You mean this console log?

![image](https://github.com/allure-framework/allure2/assets/141135676/9a1d0637-c220-4871-a424-6b122d4eab52)

**Reply by baev** [reply]
What is the size of the generated single HTML?

**Reply by huynguyen-anduin** [reply]
It's 558 MB. Is it too large?

For more info, browser still can report correctly another single HTML with size of 523.76 MB.

**Reply by baev** [reply]
I mean, the single file of 523.76 MB just takes a while to open. Maybe the browser struggles a bit opening it, give it a little more time.

But anyway, we do not recommend using single file mode for such huge reports.

**Reply by huynguyen-anduin** [reply]
To be clear, I mentioned 2 reports: 1 with size 523.76MB, and 588MB. The former can display correctly, but the later can not.

I don't think it dues to file size. I tried to open and wait many times, the report never appears.

**Reply by huynguyen-anduin** [reply]
Hi @baev, do you have any idea on what maybe the reason? Any workaround to avoid this?

**Reply by omarShalLaby** [reply]
@baev  I Have this issue also when i use the command allure generate ...any solutions for this ? i use the command `allure generate Output/allure-results --clean -o Output/allure-report` the report is displayed empty but with `mvn allure:serve` it works fine and I need the generate command so I can generate the report in git actions after the execution

**Reply by baev** [reply]
@omarShalLaby, if `$ mvn allure:serve` works, it likely means either you're providing the wrong path to the allure-results folder when using the allure generate command, or you're trying to open the report without a web server (by default, Allure requires the report to be served via a web server). In this case, you can either use the `$ allure open Output/allure-report` command or add the `--single-file` option to the generate command to switch to single-file report mode. This mode doesn't require a web server and can be opened directly from the file system in any browser. 

You can find more details in the documentation: https://allurereport.org/docs/gettingstarted-view-report/

**Reply by omarzero130** [reply]
@baev  I tried to add the`--single-file` option to the command and the whole command look like this `allure generate target/allure-results --clean -o Output/allure-report --single-file` but when it run this command locally it displays this error and the report still displayed with no data
![image](https://github.com/user-attachments/assets/80260663-9986-442c-bf08-74da05229eeb)
![image](https://github.com/user-attachments/assets/98fe1572-ebbc-45b1-8dbe-03ea74c6cc89)

**Reply by omarzero130** [reply, marked as answer]
@baev it works fine now i just needed to update the allure version to the latest(2.30.0) to use the `--single-file` option , Thanks for you support
